relay/http.go

```diff
@@ -38,8 +46,36 @@ func NewHTTP(cfg HTTPConfig) (Relay, error) {
 		if b.Name == "" {
 			b.Name = b.Location
 		}
-
-		h.backends = append(h.backends, &httpBackend{b.Name, b.Location})
+		timeout := DefaultHTTPTimeout
+		if b.Timeout != "" {
+			t, err := time.ParseDuration(b.Timeout)
+			if err != nil {
+				return nil, fmt.Errorf("error parsing HTTP timeout %v", err)
+			}
+			timeout = t
+		}
+		// If configured, create a retryBuffer per backend.
+		// This way we serialize retries against each backend.
+		var rb *retryBuffer
+		if b.BufferSize > 0 {
+			max := DefaultMaxDelayInterval
+			if b.MaxDelayInterval != "" {
+				m, err := time.ParseDuration(b.MaxDelayInterval)
+				if err != nil {
+					return nil, fmt.Errorf("error parsing max retry time %v", err)
+				}
+				max = m
+			}
+			rb = newRetryBuffer(b.BufferSize, DefaultInitialInterval, DefaultMultiplier, max)
+		}
+		h.backends = append(h.backends, &httpBackend{
+			name:        b.Name,
+			location:    b.Location,
+			retryBuffer: rb,
+			client: &http.Client{
```

Comment:
I think all the backends can safely share a single client, rather than each having their own.